pkg/plugins/diff/delta_diff_server/src/main.rs

```diff
@@ -0,0 +1,211 @@
+    }
+}
+
+fn reverse_and_create_table_ops(mut hist: Vec<Map<String, Value>>, table_version: DeltaDataTypeVersion) -> Vec<TableOperation> {
+    hist.reverse();
+    let ops = OpTypes::new();
+    let mut ans: Vec<TableOperation> = Vec::with_capacity(hist.len());
+    for m in hist {
+        let table_op = utils::construct_table_op(&m, table_version, &ops);
+        ans.push(table_op);
+    }
+    return ans;
+}
+
+fn compare(left_history_version: &mut HistoryAndVersion,
+           right_history_version: &mut HistoryAndVersion,) -> Result<Vec<TableOperation>, Error> {
+    // The lower limit of the two (earliest common version):
+    let lower_limit = right_history_version.get_lower_limit(&left_history_version);
+    let mut table_op_list: Vec<TableOperation> = vec![];
+
+    let right_table_version = right_history_version.version;
+    let left_table_version = left_history_version.version;
+    let mut curr_version = right_table_version;
+
+    // Run through the commits from newest to oldest.
+    left_history_version.history.reverse();
+    right_history_version.history.reverse();
+    let mut left_commit_slice = &left_history_version.history[..];
+    let mut right_commit_slice = &right_history_version.history[..];
+    let ops = OpTypes::new();
+
+    // If the right table's version is bigger than the left table's version, add all commits until reaching the same version as the left table.
+    if right_table_version > left_table_version {
+        let right_iter = right_history_version.history.iter();
+        for commit_info in right_iter {
+            if curr_version == left_table_version {
+                right_commit_slice = &right_history_version.history[(right_table_version - left_table_version) as usize..];
+                break;
+            }
+            match commit_info {
+                _operation_content => {
+                    let d = utils::construct_table_op(commit_info, curr_version, &ops);
+                    table_op_list.push(d)
+                }
+            }
+            curr_version -= 1;
+        }
+    } else { // If left table's version is bigger, start running over the left log from the right table's version.
+        left_commit_slice = &left_history_version.history[(left_table_version - right_table_version) as usize..];
+    }
+    compare_table_slices(left_commit_slice, right_commit_slice, curr_version, lower_limit, table_op_list)
+}
+
+fn compare_table_slices(left_commit_slice: &[Map<String, Value>],
+                        right_commit_slice: &[Map<String, Value>],
+                        mut curr_version: DeltaDataTypeVersion,
+                        lower_limit: DeltaDataTypeVersion,
+                        mut table_op_list: Vec<TableOperation>) -> Result<Vec<TableOperation>, Error> {
+    let ops = OpTypes::new();
+    let mut i: usize = 0; // iterating through the vector while 'curr_version' is the real version of the
```

Comment:
Oh man I hate it when I don't hear the punchline :(

Reply:
damn